crates/modelardb_server/src/main.rs

```diff
@@ -328,10 +327,11 @@ fn create_session_context(query_data_folder: Arc<dyn ObjectStore>) -> SessionCon
         .unwrap();
     session_runtime.register_object_store(&object_store_url, query_data_folder);
 
-    let session_state = SessionState::with_config_rt(session_config, session_runtime)
-        .with_physical_optimizer_rules(vec![Arc::new(
-            model_simple_aggregates::ModelSimpleAggregatesPhysicalOptimizerRule {},
-        )]);
+    // Use the add* methods instead of the with* methods as the with* methods replace the built-ins.
```

Comment:
Not sure I understand this comment. Maybe I just miss the big picture, but the first method call is to a with method and after that only one add method is called.

Reply:
The comment is to try and prevent the big mistake I made previously. [`SessionState`](https://docs.rs/datafusion/latest/datafusion/execution/context/struct.SessionState.html) can be extended using two sets of methods. The first set all has `with_` as a prefix, .e.g, `with_physical_optimizer_rules()`, which is simple to use but **replaces** all of the physical optimizer rules built into Apache Arrow DataFusion with ours. The second set all has `add_` as a prefix, e.g., `add_physical_optimizer_rule()`, which requires a bit more code to use but add our physical optimizer rules in addition to those already built into Apache Arrow DataFusion. I have added a link to the documentation to make the comment more clear.